src/lib/types/restrictions.ts

```diff
@@ -130,9 +131,25 @@ export const stepValidation = z
   })
   .strict()
 
-const programValidation = z.array(
-  z.object({ op: binaryOperator }).optional(),
-  z.object({ restriction: stepValidation }).optional()
-)
+const programValidationV2 = z.union([
+  z.object({
+    None: none,
+    SenderHasInputRole: senderHasInputRole.optional(),
+    SenderHasOutputRole: senderHasOutputRole.optional(),
+    OutputHasRole: outputHasRole.optional(),
+    OutputHasMetadata: outputHasMetadata.optional(),
+    InputHasRole: inputHasRole.optional(),
+    InputHasMetadata: inputHasMetadata.optional(),
+    MatchInputOutputRole: matchInputOutputRole.optional(),
+    MatchInputOutputMetadataValue: matchInputOutputMetadataValue.optional(),
+    MatchInputIdOutputMetadataValue: matchInputIdOutputMetadataValue.optional(),
+    FixedNumberOfInputs: fixedNumberOfInputs.optional(),
+    FixedNumberOfOutputs: fixedNumberOfOutputs.optional(),
+    FixedInputMetadataValue: fixedInputMetadataValue.optional(),
+    FixedOutputMetadataValue: fixedOutputMetadataValue.optional(),
+    FixedOutputMetadataValueType: fixedOutputMetadataValueType.optional(),
+  }),
+  z.object({ op: binaryOperator }),
+])
```

Comment:
Is this not just:

```ts
const programValidationV2 = z.union([
  z.object({ None: none }),
  z.object({ SenderHasInputRole: senderHasInputRole }),
  z.object({ SenderHasOutputRole: senderHasOutputRole }),
  z.object({ OutputHasRole: outputHasRole }),
  z.object({ OutputHasMetadata: outputHasMetadata }),
  z.object({ InputHasRole: inputHasRole }),
  z.object({ InputHasMetadata: inputHasMetadata }),
  z.object({ MatchInputOutputRole: matchInputOutputRole }),
  z.object({ MatchInputOutputMetadataValue: matchInputOutputMetadataValue }),
  z.object({ MatchInputIdOutputMetadataValue: matchInputIdOutputMetadataValue }),
  z.object({ FixedNumberOfInputs: fixedNumberOfInputs }),
  z.object({ FixedNumberOfOutputs: fixedNumberOfOutputs }),
  z.object({ FixedInputMetadataValue: fixedInputMetadataValue }),
  z.object({ FixedOutputMetadataValue: fixedOutputMetadataValue }),
  z.object({ FixedOutputMetadataValueType: fixedOutputMetadataValueType }),
  z.object({ op: binaryOperator }),
])
```

The point is a given object cannot contain more than one restriction type as that's not how the rust enum works